cpp/src/arrow/CMakeLists.txt

```diff
@@ -563,10 +563,10 @@ add_arrow_lib(arrow
               ${ARROW_SHARED_PRIVATE_LINK_LIBS}
               STATIC_LINK_LIBS
               ${ARROW_STATIC_LINK_LIBS}
+              ${ARROW_STATIC_INSTALL_INTERFACE_LIBS_REAL}
               SHARED_INSTALL_INTERFACE_LIBS
               ${ARROW_SHARED_INSTALL_INTERFACE_LIBS}
-              STATIC_INSTALL_INTERFACE_LIBS
-              ${ARROW_STATIC_INSTALL_INTERFACE_LIBS})
+              STATIC_INSTALL_INTERFACE_LIBS)
```

Comment:
Can we remove `STATIC_INSTALL_INTERFACE_LIBS`?

Reply:
It seems so. Changed.